### PR TITLE
[NP-5923] Medusa Failover

### PR DESCRIPTION
**Pull request by jlhughes**

### CODE REVIEW

File: src/foam/nanos/medusa/BlockingDAO.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        if ( BLOCKED_CMD.equals(obj) ) {
+          return blocked_.get();
+        }
+      }
+      return getDelegate().cmd_(x, obj);
+      `
+    },
+    {
+      name: 'block',
+      args: [
```
Comment (kgrgreer) [inline src/foam/nanos/medusa/BlockingDAO.js]: You could just say: args: 'Context x'

File: src/foam/nanos/medusa/ClusterConfigStatusDAO.js [hunk 2]
```diff
@@ -160,5 +148,46 @@ foam.CLASS({
...
+      Timer timer = new Timer(this.getClass().getSimpleName());
+      timer.schedule(
+        new AgencyTimerTask(getX(), support.getThreadPoolName(), this),
+        getInitialTimerDelay(), getTimerInterval());
+      `
+    },
+    {
+      documentation: 'Check for multiple primaries.  Can occur when the primary is issolated, the others vote, then the primary comes back - and was unaware that it was issolated.',
+      name: 'execute',
+      args: [
```
Comment (kgrgreer) [inline src/foam/nanos/medusa/ClusterConfigStatusDAO.js]: same

File: src/foam/nanos/medusa/RenouncePrimaryDAO.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  properties: [
+    {
+      name: 'serviceName',
+      class: 'String',
+      value: 'medusaEntryMediatorDAO'
+    },
+    {
+      name: 'pollingInterval',
+      class: 'Long',
+      value: 1000
```
Comment (kgrgreer) [inline src/foam/nanos/medusa/RenouncePrimaryDAO.js:54]: Maybe add units: here and for next property also.

File: src/foam/nanos/medusa/ReplayingDAO.js [hunk 4]
```diff
@@ -7,134 +7,117 @@
...
-    {
-      name: 'put_',
-      javaCode: `
-      // getLogger().debug("put");
-      blockOnReplay(x);
-      return getDelegate().put_(x, obj);
-      `
+      name: 'timerInterval',
+      class: 'Long',
+      value: 1000
```
Comment (kgrgreer) [inline src/foam/nanos/medusa/ReplayingDAO.js:33]: units:

File: src/foam/nanos/medusa/ReplayingDAO.js [hunk 5]
```diff
@@ -7,134 +7,117 @@
...
     {
-      name: 'remove_',
-      javaCode: `
-      // getLogger().debug("remove");
-      blockOnReplay(x);
-      return getDelegate().remove_(x, obj);
-      `
+      name: 'initialTimerDelay',
+      class: 'Long',
+      value: 60000
```
Comment (kgrgreer) [inline src/foam/nanos/medusa/ReplayingDAO.js:39]: units:

File: src/foam/nanos/medusa/TimeoutDAO.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  properties: [
+    {
+      name: 'enabled',
+      class: 'Boolean',
+      value: true
+    },
+    {
+      name: 'globalTimeout',
+      class: 'Long',
+      value: 30000
```
Comment (kgrgreer) [inline src/foam/nanos/medusa/TimeoutDAO.js:31]: units

**Comment by kgrgreer** [comment]
LGTM, added a few minor comments

File: src/foam/nanos/medusa/BlockingDAO.js (hunk 1 above)
Comment (jlhughes) [reply src/foam/nanos/medusa/BlockingDAO.js]: updated throughout.

**Pull request merged**